osu.Game/Rulesets/Mods/ModAutoplay.cs

```diff
@@ -30,6 +31,18 @@ public abstract class ModAutoplay : Mod, IApplicableFailOverride, ICreateReplay
 
         public override bool HasImplementation => GetType().GenericTypeArguments.Length == 0;
 
+        [Obsolete("Use CreateScoreFromReplayData(IBeatmap, IReadOnlyList<Mod>) instead")] // Can be removed 20220929
         public virtual Score CreateReplayScore(IBeatmap beatmap, IReadOnlyList<Mod> mods) => new Score { Replay = new Replay() };
+
+        public virtual ModReplayData CreateReplayData(IBeatmap beatmap, IReadOnlyList<Mod> mods)
+        {
+            Logger.Log($"Ruleset mod implementation for {GetType().Name} should be updated to newer {nameof(ICreateReplayData)} signature.", LoggingTarget.Information);
```

Comment:
Hard to give visibility that changes are required without this. Should be pretty low impact to users but high enough visibility to let ruleset creators know before support is removed.

Reply:
When I tested what this does (by rolling back the changes in one of the template rulesets), I can't say this was really visible in any way - I mean, while it logs, it does so at a verbose level and so there is no clear indication that something is wrong unless you open run output in rider, or open logs, or enable the log overlay. Was this the intention here?

Reply:
In the big picture, the point of this log is to note to ruleset developers that `CreateReplayData` should be overridden instead of `CreateReplayScore`, is that right? 

If so, I find the `ObsoleteAttribute` tag to be more than enough for a per-build warning display to all rulesets that are still overriding `CreateReplayScore`, while never visible in-game on players. Verbose-logging shouldn't be necessary for that purpose, and wouldn't give much visibility compared to the existing obsoletion tags.

Reply:
Can you show the warning you are seeing? I wasn’t seeing any.

@bdach The log call should have been Important level. Somehow I didn’t commit that, probably from stashing too much.

Reply:
Weird, this didn't seem to show when i was testing. I'll remove the log output as this should already be loud enough.